src/passes/pass.cpp

```diff
@@ -723,6 +723,12 @@ void PassRunner::addDefaultGlobalOptimizationPostPasses() {
   }
   // may allow more inlining/dae/etc., need --converge for that
   addIfNoDWARFIssues("directize");
+
+  // Translate to use the new EH instructions if requested.
+  if (wasm->features.hasExceptionHandling() && options.experimentalNewEH) {
```

Comment:
Doing this here means it will run twice if the user does `wasm-opt -O3 -O3`. Perhaps it would be better to insert this pass in `src/tools/wasm-opt.cpp`? Then we can add it right at the end.

Reply:
A downside to my idea is that we'd need to also add logic there for adding the stackIR passes right after it, copying the 4 lines underneath us. But I think it might still be better.

Reply:
Then if a user gives `wasm-opt --generate-stack-ir --optimize-stack-ir --experimental-new-eh` (or `-O3`, which I guess also runs the stack IR generation and optimization), aren't we gonna generate and optimize the stack IR, throw it away, run the translator, and re-generate and re-optimize the stack IR?

Reply:
Yes. We also do that with `-O3 -O3`, as the end of each of them generates stack IR and optimizes it. A little wasteful, but not significant (making the pass pipeline smarter to avoid this overhead likely isn't worth it).

Reply:
Is `-O3 -O3` common enough? I thought that would only happen if someone carelessly add it twice, not remembering they have already done it, but I might be wrong.. whereas I thought `-O3 --experimental-new-eh` would be pretty common, because it's basically just `-O3` with the intention of using the new EH, which now runs stack IR generation/optimization twice.

Reply:
Chatted w/ @tlively and I learned people run `-O3` not only twice but many times... Yeah, it'd be better to run it only once at the end of `wasm-opt`, even though it means running Stack IR twice.

Reply:
Closing this in favor of #6270, which adds the option directly to `wasm-opt.cpp`.